### DISCUSSION TITLE
Aboud MSC in regression.cpp

### DISCUSSION THREAD

**Discussion by Theotaeho** [sandialabs/UQTk#10] [Q&A]
Hi, I am a new user of UQTk (Downloading github file Apr. 3, 2024). 
I downloaded this in WSL2-Ubuntu 22.04

I sucessfully downloaded UQTk and binded UQTk with Python.  
Every test was passed. 

After making 'UQTk-install', I connected this to Jupyter lab. 
The problem was that I got some errors for *.ipynb files in the folder, "examples/dfi_app/", especially in this part. 
**"subprocess.run(shlex.split(f"{path_to_python} {path_to_uq_pc} -r offline_post -p param_ranges.dat -m lsq -n {xs.shape[0]} -s rand -t 2 -z"), cwd=tmp, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)"**
When I run this, the error message showed me that 
"[Errno 2] No such file or directory: '/home/theokim/UQTk-install/examples/dfi_app/results.pk'"

When I visit the "uq_pc.py" file, the initial "pred_mode" was "ms". 
Here, my problem can be solved after changing initial  "pred_mode" from "ms" - mean + std to "m" - mean.

If I run "regression" using terminal using "ms" or "msc", 
./regression -l0 -x /home/theokim/UQTk-install/examples/dfi_app/xdata.dat -y /home/theokim/UQTk-install/examples/dfi_app/ydata.dat -b PC -p /home/theokim/UQTk-install/examples/dfi_app/mi.dat -m ms
or
./regression -l0 -x /home/theokim/UQTk-install/examples/dfi_app/xdata.dat -y /home/theokim/UQTk-install/examples/dfi_app/ydata.dat -b PC -p /home/theokim/UQTk-install/examples/dfi_app/mi.dat -m msc

The terminal output was like below. And the results showed **"Segmentation fault"** in the code.:
############
xfile      = /home/theokim/UQTk-install/examples/dfi_app/xdata.dat 
yfile      = /home/theokim/UQTk-install/examples/dfi_app/ydata.dat 
basistype  = PC 
method     = lsq 
intpar     = 5 
dblpar     = 0 
strpar     = LU 
msc        = msc 
Dimensionality 3
Number of bases 56
Segmentation fault
############

If I run "regression" using terminal using "m", 
############
xfile      = /home/theokim/UQTk-install/examples/dfi_app/xdata.dat 
yfile      = /home/theokim/UQTk-install/examples/dfi_app/ydata.dat 
basistype  = PC 
method     = lsq 
intpar     = 5 
dblpar     = 0 
strpar     = LU 
msc        = m 
Dimensionality 3
Number of bases 56
LOO error : 7.85189e-08
GCV error : -5.06816e-09
############

I think there are some problems inside of the source code when running the "ms" or "msc" options. 
Can you fix it for this? If this is fixed, it would be much helpful for using this !

Thank you so much for your help!

Taeho Kim

**Comment by bjdebus** [comment]
Hi @Theotaeho , I am seeing this discussion here but I also vaguely remember talking about this over email. Has this been resolved, or am I confusing this with a different issue? As far as I know, this bug has been fixed now.

**Comment by Theotaeho** [comment]
Yes, this issue was already solved.

Thanks for checking!

I have one quick question. Is there some future plan making some python
wrapper of UQTK to PYMC?

If we can use MCMC from PYMC to UQTK, it would be very powerful.

Thanks!

Taeho Kim.

**Reply by bjdebus** [reply]
Hi @Theotaeho , glad to hear.

Regarding your question about PYMC and UQTk, are you looking to use MCMC solvers in PYMC on models that are implemented in UQTk? If so, I think that should be possible using the current Python interface to UQTk. If you are looking to use UQTk MCMC samplers for models in PYMC, that would be a different story. We have not put python wrappers around our C++ MCMC samplers yet and it is not in the plans for the near future.

Hope this helps,
Bert
